lighthouse-extension/app/src/lighthouse-ext-background.js

```diff
@@ -111,14 +112,29 @@ async function runLighthouseInLR(connection, url, flags, {lrDevice, categoryIDs,
     config.settings.onlyCategories = categoryIDs;
   }
 
-  const results = await lighthouse(url, flags, config, connection);
-  if (!results) return;
+  try {
+    const results = await lighthouse(url, flags, config, connection);
+    if (!results) return;
 
-  if (logAssets) {
-    await assetSaver.logAssets(results.artifacts, results.lhr.audits);
-  }
+    if (logAssets) {
+      await assetSaver.logAssets(results.artifacts, results.lhr.audits);
+    }
+    return results.report;
+  } catch (err) {
+    // If an error ruined the entire lighthouse run, attempt to return a meaningful error.
+    if (!(err instanceof LHError) || !err.lhrRuntimeError) {
+      throw err;
```

Comment:
In this situation I think we should not throw but instead return an "UNKNOWN_ERROR" i guess.

tbh i dont know what throwing does in this context. my guess is its not caught and triggers a timeout?

Reply:
done